native-static-images/README.md

```diff
@@ -0,0 +1,111 @@
+This approach is ideal for deployment on a distroless container image.
+
+> Note: This currently only works when linked against `libc`.
+
+This guide shows how you can take advantage of Native Image linking options including fully dynamic, fully static, and mostly static (except `libc`) to generate an executable ideal for your deployment scenario.
+
+For the following guide you should be have:
+
+- Linux AMD64 operating system
+- GraalVM distribution for Java 11 or higher.
+- A 64-bit `musl` toolchain, `make`, and `configure`
+- The latest `zlib` library
+
+## Preparation
+
+1. Download and install the latest GraalVM JDK with Native Image using the [GraalVM JDK Downloader](https://github.com/graalvm/graalvm-jdk-downloader):
+    ```bash
+    bash <(curl -sL https://get.graalvm.org/jdk) -c 'native-image' 
+    ```
+
+As a preparation step, you need install the `musl` toolchain, compile and install `zlib` into the toolchain.
+
+2. Download the `musl` toolchain from [musl.cc](https://musl.cc/). (We recommend [this one](https://more.musl.cc/10/x86_64-linux-musl/x86_64-linux-musl-native.tgz). Extract the toolchain to a directory of your choice. This directory will be referred as `$TOOLCHAIN_DIR`. 
+
+    Download the latest `zlib` library sources from [zlib.net](https://zlib.net/) and extract them. (This documentation uses `zlib-1.2.11`.)
+    Create a new environment variable, named `CC`:
+    ```bash
+    CC=$TOOLCHAIN_DIR/bin/gcc
+    ```
+
+    Change into the `zlib` directory, and then run the following commands to compile and install `zlib` into the toolchain:
+    ```bash
+    ./configure --prefix=$TOOLCHAIN_DIR --static
+    ```
+    ```bash
+    make
+    ```
+    ```bash
+    make install
+    ```
+
+3. Download or clone the repository and navigate into the `native-static-images` directory:
+    ```bash
+    git clone https://github.com/graalvm/graalvm-demos
+    ```
+    ```bash
+    cd graalvm-demos/native-static-images
+    ```
+
+## Build a Static Native Executable
+
+The source code saved in _EnvMap.java_ iterates over your environment variables and prints out the ones that contain the `String` of characters passed as a command line argument.
+
+1. First, ensure the directory named `$TOOLCHAIN_DIR/bin` is present on your `PATH`.
+    To verify this, run the following command:
+    ```bash
+    x86_64-linux-musl-gcc
+    ```
+    You should see output similar to the following:
+    ```
+    x86_64-linux-musl-gcc: fatal error: no input files
+    compilation terminated.
+    ```
+2. Compile the file:
+    ```shell
+    $JAVA_HOME/bin/javac EnvMap.java
+    ```
+
+3. Build a static native executable by running this command:
+    ```shell
+    $JAVA_HOME/bin/native-image --static --libc=musl EnvMap
+    ```
+    This produces a native executable with statically linked system libraries.
```

Comment:
This produces a native executable with statically linked system libraries (including JDK-shared libraries).